packages/ui/react-ui/styles.css

```diff
@@ -1,260 +1,383 @@
+        margin: 1rem;
+        min-height: calc(100vh - 2rem); /* 100vh - 2 * margin */
+    }
 }
 
-.wallet-adapter-button-end-icon,
-.wallet-adapter-button-start-icon,
-.wallet-adapter-button-end-icon img,
-.wallet-adapter-button-start-icon img {
+.wallet-modal-h-52 {
+    height: 52rem !important;
+}
+.wallet-modal-pb-2 {
+    padding-bottom: 2rem !important;
+}
+
+.wallet-adapter-modal-wrapper {
+    height: 45rem;
+    box-sizing: border-box;
+    position: relative;
     display: flex;
     align-items: center;
+    flex-direction: column;
+    z-index: 1050;
+    max-width: 24rem;
+    width: 100%;
+    border-radius: 0.5rem;
+    background: #10141f;
+    font-family: 'DM Sans', 'Roboto', 'Helvetica Neue', Helvetica, Arial, sans-serif;
+    font-weight: 500;
+    flex: 1;
+    padding: 0;
+    transition: height .5s ease;
+}
+
+.wallet-adapter-modal-button-close {
+    position: absolute;
+    right: 1rem;
+    top: 1rem;
+    display: flex;
+    height: 2rem;
+    width: 2rem;
+    border-radius: 50%;
     justify-content: center;
-    width: 24px;
-    height: 24px;
+    align-items: center;
+    cursor: pointer;
+    background-color: rgba(255, 255, 255, 0.05);
+    border: none;
+    font-family: 'DM Sans', 'Roboto', 'Helvetica Neue', Helvetica, Arial, sans-serif;
 }
 
-.wallet-adapter-button-end-icon {
-    margin-left: 8px;
+.wallet-adapter-modal-button-close svg {
+    fill: rgba(255, 255, 255, 0.7);
 }
 
-.wallet-adapter-button-start-icon {
-    margin-right: 8px;
+.wallet-adapter-modal-button-close:focus-visible {
+    outline-color: white;
 }
 
-.wallet-adapter-collapse {
-    width: 100%;
+.wallet-adapter-modal-button-close:hover > svg {
+    fill: rgba(255, 255, 255, 1);
+    transition: fill 0.1s ease;
 }
 
-.wallet-adapter-dropdown {
-    position: relative;
-    display: inline-block;
+.wallet-adapter-modal-title {
+    font-size: 1.75rem;
+    line-height: 2.1rem;
+    color: #fff;
+    margin: 3.5rem 0 0 0;
+    font-weight: 700;
+    text-align: center;
+    padding: 0 2rem;
 }
 
-.wallet-adapter-dropdown-list {
-    position: absolute;
-    z-index: 99;
-    display: grid;
-    grid-template-rows: 1fr;
-    grid-row-gap: 10px;
-    padding: 10px;
-    top: 100%;
-    right: 0;
-    margin: 0;
+@media (max-width: 26.875em) {
+    .wallet-adapter-modal-title {
+        padding: 0 1rem;
+    }
+}
+
+.wallet-adapter-modal-list-header {
+    margin: 5rem 0rem 0rem 1.6875rem;
+    font-size: 1rem;
+    line-height: 1.5rem;
+    font-weight: 500;
+    align-self: flex-start;
+}
+
+.wallet-modal-mt-24 {
+    margin-top: 1.5rem !important;
+}
+
+.wallet-adapter-modal-list {
+    margin: 3rem 0 0 0;
+    padding: 0.125rem;
     list-style: none;
-    background: #2c2d30;
-    border-radius: 10px;
-    box-shadow: 0px 8px 20px rgba(0, 0, 0, 0.6);
-    opacity: 0;
-    visibility: hidden;
-    transition: opacity 200ms ease, transform 200ms ease, visibility 200ms;
+    width: 100%;
+    max-height: 17.5rem;
+    overflow-y: auto;
```

Comment:
The sizing of the scrolling list is such that a user may be very likely to miss the options below the fold:

<img width="346" alt="Screen Shot 2021-12-15 at 5 59 59 PM" src="https://user-images.githubusercontent.com/1173161/146283541-4fae86b4-06c2-4e32-9f56-a3504141300c.png">

It's not obvious that you can scroll this list and there are more wallets.